metis/bin/metis_client

```diff
@@ -1278,21 +1299,21 @@ EOT
 
       # compute md5sums
       files.each.with_index do |file,i|
-        file = Pathname.new(file).cleanpath.to_s
+        file = clean_path(file)
 
         print "Computing md5s - #{progress_bar(i.to_f / files.length)} - #{i} of #{files.length}      \r"
 
         next if ::File.basename(file) == options[:file] || md5_cache[file]
 
-        md5_cache[file] = %x{ md5sum "#{file}" }.split(' ')[0]
+        md5_cache[file] = %x{ md5sum #{Shellwords.escape(file)} }.split(' ')[0]
       end
       puts "Computing md5s - #{progress_bar(1)} - #{files.length} of #{files.length}      "
 
-      File.open("md5checksum.txt", "w") do |f|
-        output =  md5_cache.map do |file_name, checksum|
-          "#{checksum}  #{file_name}"
-        end.join("\n")
+      output =  md5_cache.map do |file_name, checksum|
+        "#{checksum}  #{file_name}"
+      end.join("\n")
```

Comment:
If output computation fails/errors for some reason, this could previously blank the md5checksum.txt file, which represents many precious hours of computation.